src/test/java/gov/cabinetoffice/gap/adminbackend/services/GrantAdvertServiceTest.java

```diff
@@ -120,6 +124,42 @@ void create_success() {
 
                 when(grantAdminRepository.findById(grantAdminId)).thenReturn(Optional.of(grantAdmin));
                 when(schemeRepository.findById(grantSchemeId)).thenReturn(Optional.of(grantScheme));
+                when(featureFlagsConfigurationProperties.isNewMandatoryQuestionsEnabled()).thenReturn(false);
+                when(grantAdvertRepository.save(any())).thenAnswer(i -> {
+                    GrantAdvert output = (GrantAdvert) i.getArguments()[0];
+                    output.setId(id);
+                    return output;
+                });
+
+                GrantAdvert outputAdvert = grantAdvertService.create(grantSchemeId, grantAdminId, name);
+
+                assertThat(outputAdvert).isEqualTo(expectedAdvert);
+            }
+
+        }
+
+        @Test
+        void create_successWhenNewMandatoryQuestionFeatureFlagIsTrue() {
+            final String instantExpected = "2014-12-22T10:15:30Z";
+            final Clock clock = Clock.fixed(Instant.parse(instantExpected), ZoneId.of("UTC"));
+            final Instant instant = Instant.now(clock);
+            final UUID id = UUID.randomUUID();
+            final int grantAdminId = 1;
+            final int grantSchemeId = 1;
+            final String name = "Test Grant Advert";
+            final GrantAdmin grantAdmin = GrantAdmin.builder().id(grantAdminId)
+                    .funder(FundingOrganisation.builder().id(1).build()).build();
+            final SchemeEntity grantScheme = SchemeEntity.builder().id(grantSchemeId).funderId(1).build();
+
+            try (MockedStatic<Instant> mockedStatic = mockStatic(Instant.class)) {
```

Comment:
Its not something we need to change just now but just as an FYI the `Instant.now` method also has an overload that accepts a clock. Thats a bit of a nicer way to handle static creation of times and dates in Java as you can pass in a fixed clock object and don't need to use these horrible `mockedStatic` calls. 

It seems like we actually use it in the test a few lines above this but the change was never made inside the service class.